### PR TITLE
Adds `last_index` and `clear` methods

### PR DESCRIPTION
**Pull request by e-ivkov**
## Future improvements
1. Return `Option::None` from `last_index` and `first_index` in case there are no entries
For now I kept the interface from `first_index` and return `0` in that case
2. Move out common logic in quickcheck tests

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -621,6 +636,68 @@ mod test {
...
+            let entries = EntryGenerator::new()
+                .into_iter()
+                .take(entry_count as usize)
+                .collect::<Vec<_>>();
+
+            for entry in &entries {
+                wal.append(entry).unwrap();
+            }
+            let last_index = wal.last_index();
+            if wal.entry(last_index).is_none() && wal.num_entries() != 0 {
```
Comment (agourlay) [inline src/lib.rs:661]: Why do we need `&& wal.num_entries() != 0` ?
Is it because `entry_count` can be zero via Quickcheck?

Comment (e-ivkov) [reply src/lib.rs:661]: Yes, and in that case entry will be none

Comment (agourlay) [reply src/lib.rs:661]: nitpick: If there is no value in testing `entry` equals to zero, maybe we could return early with `return TestResult::discard();` in that case

Comment (e-ivkov) [reply src/lib.rs:661]: We still want to test that `last_index` will not panic

Comment (e-ivkov) [reply src/lib.rs:661]: It actually did when there was no zero check and it overflowed

**Review by agourlay — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (agourlay) [reply src/lib.rs:661]: let's keep it then!

**Pull request merged**